### PR TITLE
Add protein sequence information to transcript view

### PR DESCRIPTION
**Pull request by mdml**
* Add back (optional) protein sequence to transcript view, which appears (in front of any domains) when you zoom in far enough.
* Add new (optional) protein sequence annotations, which are represented as colored letters from the 
* Add a protein sequence annotation legend, which is an inline list from Bootstrap.
* Add back the ability to switch protein domains.
* Update `demo/test-transcript.html` to demonstrate these new features.

### CODE REVIEW

**Comment by connorgr** [comment]
Overall the changes look good to me. I only saw three minor things that need to be changed:

1. Protein sequence font family is serif'd rather than the default sans-serif fonts defined in the transcript style file.
2. I'm not sure if this is a bug with the transcript plot demos, but the vertical slider for viewing clipped point mutation stacks isn't present.
3. I'd also prefer to remove any Bootstrap dependencies in GD3 – I don't think there are any right now, though I could be wrong. Inline list styling should only be ~5 lines of CSS.

File: src/transcript/transcriptChart.js [hunk 1]
```diff
@@ -159,14 +162,43 @@ function transcriptChart(style) {
...
+          seqAnnotationColorRange.push(style.seqAnnotationColors[anno_ty])
+        } else{
+          seqAnnotationColorRange.push(defaultColors[defaultColorIndex]);
+          defaultColorIndex += 1;
+        }
+      })
+      var seqAnnotationColor = d3.scale.ordinal().domain(data.seq_annotation_types).range(seqAnnotationColorRange);
+
+      // Add protein sequence information, if it's present
+      if (data.sequence){
```
Comment (connorgr) [inline src/transcript/transcriptChart.js]: space missing between ) and {

File: src/transcript/transcriptChart.js [hunk 2]
```diff
@@ -159,14 +162,43 @@ function transcriptChart(style) {
...
 
-
-      updateTranscript();
+      // Set up a color scale for the sequence annotations, using any given color mapping
+      // and filling in default colors for the other annotation types
+      var seqAnnotationColorRange = [],
+          defaultColors = d3.scale.category10().range(),
+          defaultColorIndex = 0;
+
+      data.seq_annotation_types.forEach(function(anno_ty){
```
Comment (connorgr) [inline src/transcript/transcriptChart.js]: space missing between ) and {

File: src/transcript/transcriptChart.js [hunk 3]
```diff
@@ -159,14 +162,43 @@ function transcriptChart(style) {
...
+        if (anno_ty in style.seqAnnotationColors){
```
Comment (connorgr) [inline src/transcript/transcriptChart.js]: space missing between ) and {

File: src/transcript/transcriptChart.js [hunk 4]
```diff
@@ -159,14 +162,43 @@ function transcriptChart(style) {
...
+          seqAnnotationColorRange.push(style.seqAnnotationColors[anno_ty])
+        } else{
```
Comment (connorgr) [inline src/transcript/transcriptChart.js]: space missing between ) and {

File: src/transcript/transcriptChart.js [hunk 5]
```diff
@@ -159,14 +162,43 @@ function transcriptChart(style) {
...
+      // Add protein sequence information, if it's present
+      if (data.sequence){
+        var seq = tG.append('g')
+          .attr('class', 'gd3ProteinSequence')
+          .attr('transform', 'translate(0,' + (style.height/2 + style.transcriptBarHeight/2 + 6) + ')')
+          .selectAll('.seq')
+          .data(data.sequence).enter()
+          .append('text')
+          .attr('text-anchor', 'middle')
+          .style('fill', function(d, i){
```
Comment (connorgr) [inline src/transcript/transcriptChart.js]: space missing between ) and {

File: src/transcript/transcriptChart.js [hunk 6]
```diff
@@ -304,7 +335,16 @@ function transcriptChart(style) {
             return w/2;
           } else { return d3.select(this).attr('x'); }
         });
+
+        // Update protein sequence position and visibility
+        if (data.sequence){
```
Comment (connorgr) [inline src/transcript/transcriptChart.js]: space missing between ) and {

**Comment by mdml** [comment]
I fixed the stylistic stuff -- inlining all the style, fixing the font, and fixing the code style. Good catch on the scrollers! Those were hidden, which was definitely a bug.

**Pull request merged**